src/clj/dashboard_clj/routes.clj

```diff
@@ -23,15 +25,79 @@
   (GET "/version" _
     {:status 200
      :headers {"Content-Type" "text/json; charset=utf-8"}
-     :body (str (json/write-str {:version (version/get-version
-                                            "vanilla"
-                                            "vanilla"
-                                            "version number not found")}))})
+     :body (do
+             (prn "get version")
+             (str (json/write-str {:version (version/get-version
+                                              "vanilla"
+                                              "vanilla"
+                                              "version number not found")})))})
 
   (GET "/services" _
     {:status 200
      :headers {"Content-Type" "text/json; charset=utf-8"}
-     :body (str (json/write-str {:services (h/get-services)}))})
+     :body (do
+             ;(prn "get services")
+             (str (json/write-str {:services (h/get-services)})))})
+
+;;;;; Layout Saving Routes
+
+  ;TODO test removing the write-str here and put the de-string back in url handler
+  (GET "/layout" req
+    {:status 200
+     :headers {"Content-Type" "text/json; charset=utf-8"}
+     :body (do
+             ;(prn "get layout")
+             (str (json/write-str {:layout (h/get-layout (get-in req [:params :username]))})))})
+
+  (POST "/save-layout" req
+    {:status 200
+     :headers {"Content-Type" "text/json; charset=utf-8"}
+     :body (do
+             ;(prn "save layout")
+             {:widget-save (h/save-layout (get-in req [:params :widgets]))})})
+
+  (POST "/update-widget" req
+    {:status 200
+     :headers {"Content-Type" "text/json; charset=utf-8"}
+     :body (do
+             ;(prn "updating one widget" (get-in req [:params :widget]))
+             {:widget-update (h/update-widget (get-in req [:params :widget]))})})
+
+  (POST "/delete-widget" req
+    {:status 200
+     :headers {"Content-Type" "text/json; charset=utf-8"}
+     :body (do
+             ;(prn "remove widget" (get-in req [:params :id]))
+             {:delete-widget (h/delete-widget (get-in req [:params :id]))})})
+
+;;;;;; User Login Routes
+
+  (GET "/verify-user" req
+    {:status 200
+    :headers {"Content-Type" "text/json; charset=utf-8"}
```

Comment:
the "user" routes are not indented correct, so parinfer may cause problems down the road. please push the :headers and :body segments right so they line up with :status "inside" the {}-map